packages/core/core/src/BundleGraph.js

```diff
@@ -294,6 +300,58 @@ export default class BundleGraph {
       });
   }
 
+  getBundleGroupsReferencedByBundle(
+    bundle: Bundle
+  ): Array<{bundleGroup: BundleGroup, dependency: IDependency}> {
+    let node = nullthrows(
+      this._graph.getNode(bundle.id),
+      'Bundle graph must contain bundle'
+    );
+
+    let groupNodes: Array<BundleGroupNode> = [];
+    this._graph.traverse(
+      (node, context, actions) => {
+        if (node.type === 'bundle_group') {
+          groupNodes.push(node);
+          actions.skipChildren();
+        }
+      },
+      node,
+      'bundle'
+    );
+
+    return (
+      groupNodes
+        .map(groupNode => {
+          return {
+            groupNode,
+            dependencyNode: this._graph
+              .getNodesConnectedTo(groupNode)
+              .find(
+                node =>
+                  node.type === 'dependency' &&
+                  this._graph.hasEdge(bundle.id, node.id, 'contains')
+              )
+          };
+        })
+        // TODO: Remove this filter when bundle groups have the correct bundles
+        // pointing to them
+        .filter(
```

Comment:
This filter works around a bug where bundles up the tree don't get prior connections removed during the bundling process. I think this was pointed out here: https://github.com/parcel-bundler/parcel/pull/3259/files#r306637515